packages/web3-eth/src/index.ts

```diff
@@ -0,0 +1,374 @@
+// Disabling because returnTypes must be last param to match 1.x params
+/* eslint-disable default-param-last */
+
+import { TransactionCall, TransactionWithSender, Web3BaseProvider } from 'web3-common';
+import { Web3Config, Web3RequestManager } from 'web3-core';
+import {
+	BlockNumberOrTag,
+	ValidTypes,
+	ValidReturnTypes,
+	convertToValidType,
+	Address,
+	Uint256,
+	HexString32Bytes,
+	HexStringBytes,
+	Uint,
+	HexString8Bytes,
+	convertObjectPropertiesToValidType,
+	isHexString32Bytes,
+	Filter,
+} from 'web3-utils';
+import {
+	convertibleBlockProperties,
+	convertibleFeeHistoryResultProperties,
+	convertibleReceiptInfoProperties,
+	convertibleTransactionInfoProperties,
+} from './convertible_properties';
+
+import * as rpcMethods from './rpc_methods';
+
+export class Web3Eth {
```

Comment:
according to 1.x , web3.eth should also have following functions exposed:
setProvider, given providers, current Provider, ( all these functions should be wrapper to web3Context if you are using instance, else if using inheritance these should be visible to user)

Reply:
Its also missing exposing ( batchRequest and defaultConfig items to user according to 1.x ( defaultAccount, defaultBlock, ..etc )

Reply:
> ...setProvider, given providers, current Provider...

Will be fixed when `Web3Context` is used.

Default config is provided [here](https://github.com/ChainSafe/web3.js/pull/4525/files/f1b956264f03648d780b9c69926b422853c2a99f#diff-3113e3877d2e33382025ff60b3d62ce1eb326e60e9ac21cb4162d9577b7d6784R36) and subsequently used like [so](https://github.com/ChainSafe/web3.js/pull/4525/files/f1b956264f03648d780b9c69926b422853c2a99f#diff-3113e3877d2e33382025ff60b3d62ce1eb326e60e9ac21cb4162d9577b7d6784R95)

I'm not sure about `batchRequest` I think it should be apart of `requestManager`, but I need to look into this

Reply:
You are right, I have same understanding, `batchRequest` should not be part of Eth package but it should be exposed/exported from eth package. so we maintain 1.x feature parity.